src/parser/tokenizer.rs

```diff
@@ -263,13 +264,36 @@ impl<'a> Tokenizer<'a> {
             => false,
             _ => !c.is_whitespace(),
         };
-        let (_, mut vec) = self.input.take_while(fun).map_err(to_token_error)?;
-        vec.insert(0, ch);
+        let (_, vec) = self.input.take_while(fun).map_err(to_token_error)?;
         Ok(Token::Key(pos, vec))
     }
 
-    pub fn next_token(&mut self) -> Result<Token, TokenError> {
-        let (pos, ch) = self.input.next_char().map_err(to_token_error)?;
+    pub fn next_token(&mut self) -> Result<Token<'a>, TokenError> {
+        let (pos, ch) = self.input.peek_char().map_err(to_token_error)?;
+
+        match ch {
+            CH_DOT |
+            CH_ASTERISK |
+            CH_LARRAY |
+            CH_RARRAY |
+            CH_LPAREN |
+            CH_RPAREN |
+            CH_AT |
+            CH_QUESTION |
+            CH_COMMA |
+            CH_SEMICOLON |
+            CH_SINGLE_QUOTE |
+            CH_DOUBLE_QUOTE |
+            CH_EQUAL |
+            CH_GREATER |
+            CH_LITTLE |
+            CH_AMPERSAND |
+            CH_PIPE |
+            CH_EXCLAMATION |
+            ' ' => {self.input.next_char().unwrap();},
+            _ => {},
+        }
+
         match ch {
             CH_DOLLA => self.dolla(pos, ch),
```

Comment:
I was wondering about this `DOLLA` / `dolla` spelling... do you know if this is a typo or on purpose? maybe we should fix it?

Reply:
Not sure, this is how it is on the main repository..